frame/ethereum/src/lib.rs

```diff
@@ -738,6 +771,29 @@ impl<T: Config> Pallet<T> {
 
 		let is_transactional = true;
 		let validate = false;
+
+		let (transaction_len, weight_limit) =
+			match <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+				gas_limit.low_u64(),
+				true,
+			) {
+				weight_limit if weight_limit.proof_size() > 0 => {
+					// Try to subtract the encoded extrinsic from the Weight proof_size limit or fail
+					// Validate the weight limit can afford recording transaction len
+					let _ = weight_limit
+						.proof_size()
+						.checked_sub(Self::transaction_len(transaction))
+						.ok_or(DispatchErrorWithPostInfo {
+							post_info: PostDispatchInfo {
+								actual_weight: None,
```

Comment:
Shouldn't we return some weight here?

Reply:
Maybe something equivalent to `21_000` gas or some other safe minimum. Same with the `pays_fee:` below, it's important in any unexpected cases because otherwise it might make it possible to abuse without charging any fee and accounting for any weight (meaning there's no bound on the number that can be included in a block).